library/src/main/cpp/decoder_jpeg.cpp

```diff
@@ -71,15 +72,64 @@ ImageInfo JpegDecoder::parseInfo() {
   };
 }
 
+cmsHPROFILE JpegDecoder::getColorProfile(jpeg_decompress_struct* jinfo) {
+  JOCTET* icc_data;
+  unsigned int icc_size;
+  if (jpeg_read_icc_profile(jinfo, &icc_data, &icc_size)) {
+    cmsHPROFILE src_profile = cmsOpenProfileFromMem(icc_data, icc_size);
+    free(icc_data);
+
+    cmsColorSpaceSignature profileSpace = cmsGetColorSpace(src_profile);
+
+    if (profileSpace != cmsSigRgbData &&
+        (jinfo->jpeg_color_space != JCS_GRAYSCALE ||
+         profileSpace != cmsSigGrayData)) {
```

Comment:
Are these effectively specifically supported color profiles?

Reply:
For jpeg, yes. It's either RGB image + rgb profile, or grayscale image + grayscale profile. If I recall it's just a mismatch thing. https://searchfox.org/mozilla-central/source/image/decoders/nsJPEGDecoder.cpp#326